third_party/move/testing-infra/transactional-test-runner/src/framework.rs

```diff
@@ -675,29 +698,26 @@ impl<'a> CompiledState<'a> {
             "Error publishing module: '{}'. \
              Re-publishing modules in pre-compiled lib is not yet supported",
             id
-        )
+        );
     }
 }
 
 fn compile_source_unit_v2(
-    pre_compiled_deps: Option<&FullyCompiledProgram>,
+    pre_compiled_deps: Option<&PrecompiledFilesModules>,
     named_address_mapping: BTreeMap<String, NumericalAddress>,
     deps: &[String],
     path: String,
     known_attributes: &BTreeSet<String>,
     language_version: LanguageVersion,
     experiments: Vec<(String, bool)>,
-) -> Result<(
-    (Option<CompiledModule>, Option<CompiledScript>),
-    Option<String>,
-)> {
+) -> Result<(AnnotatedCompiledUnit, Option<String>)> {
     let deps = if let Some(p) = pre_compiled_deps {
-        // The v2 compiler does not (and perhaps never) supports precompiled programs, so
-        // compile from the sources again, computing the directories where they are found.
+        // The v2 compiler does not really support precompiled programs, so we must include all the
```

Comment:
Not an action item for this PR, but wanted to understand:
(1) why does v2 compiler not really support precompiled programs (whereas, I assume, v1 does)?
(2) can this be a source of performance inefficiencies when using v2?